google/cloud/security/scanner/scanner.py

```diff
@@ -85,81 +75,47 @@
 OUTPUT_TIMESTAMP_FMT = '%Y%m%dT%H%M%SZ'
 
 
-def main(_):
-    """Run the scanner."""
-    if FLAGS.list_engines is True:
-        _list_rules_engines()
-        sys.exit(1)
-
-    if not FLAGS.engine_name:
-        LOGGER.warn('Provide an engine name')
-        sys.exit(1)
-    else:
-        rules_engine_name = FLAGS.engine_name
-
-    LOGGER.info('Using rules engine: %s', rules_engine_name)
-
-    LOGGER.info('Initializing the rules engine:\nUsing rules: %s', FLAGS.rules)
-
-    if not FLAGS.rules:
-        LOGGER.warn(('Provide a rules file. '
-                     'Use "forseti_scanner --helpfull" for help.'))
-        sys.exit(1)
-
-    forseti_config = FLAGS.forseti_config
-    if forseti_config is None:
-        LOGGER.error('Path to Forseti Security config needs to be specified.')
-        sys.exit()
-
-    try:
-        configs = file_loader.read_and_parse_file(forseti_config)
-    except IOError:
-        LOGGER.error('Unable to open Forseti Security config file. '
-                     'Please check your path and filename and try again.')
-        sys.exit()
-    configs = file_loader.read_and_parse_file(forseti_config)
-    global_configs = configs.get('global')
-    scanner_configs = configs.get('scanner')
-
-    snapshot_timestamp = _get_timestamp(global_configs)
-    if not snapshot_timestamp:
-        LOGGER.warn('No snapshot timestamp found. Exiting.')
-        sys.exit()
-
-    # Load scanner from map
-    scanner = sm.SCANNER_MAP[rules_engine_name](global_configs,
-                                                snapshot_timestamp)
-
-    # The Groups Scanner uses a different approach to apply and
-    # evaluate the rules.  Consolidate on next scanner design.
-    if rules_engine_name == 'GroupsRulesEngine':
-        all_violations = scanner.run(FLAGS.rules)
-        resource_counts = None
-    else:
-        # Instantiate rules engine with supplied rules file
-        rules_engine = em.ENGINE_TO_DATA_MAP[rules_engine_name](
-            rules_file_path=FLAGS.rules, snapshot_timestamp=snapshot_timestamp)
-        rules_engine.build_rule_book(global_configs)
-
-        iter_objects, resource_counts = scanner.run()
-
-        # Load violations processing function
-        all_violations = scanner.find_violations(
-            itertools.chain(
-                *iter_objects),
-            rules_engine)
-
-    # If there are violations, send results.
-    flattening_scheme = sm.FLATTENING_MAP[rules_engine_name]
-    if all_violations:
-        _output_results(global_configs,
-                        scanner_configs,
-                        all_violations,
-                        snapshot_timestamp,
-                        resource_counts=resource_counts,
-                        flattening_scheme=flattening_scheme)
-
-    LOGGER.info('Scan complete!')
+def _get_runnable_scanners(global_configs, scanner_configs, snapshot_timestamp):
```

Comment:
Does it make sense to have some logic that wouldn't instantiate all scanners, just the ones enabled in the scanner?
Also, your readability and ease of development, would it make sense to use a similar approach to the one used for pipelines [https://github.com/GoogleCloudPlatform/forseti-security/blob/51f48bc01bb056dc61e3722ca60761f14136d0f8/google/cloud/security/inventory/pipeline_requirements_map.py](url)